### PR TITLE
Feat/7713 analytics

### PR DESCRIPTION
**Pull request by morhag90**
Close: #7713 

## PR Details

Add analytics to custom-modules/actions/dtos services. 

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-server/src/core/module/module.service.ts [hunk 1]
```diff
@@ -61,13 +64,55 @@ export class ModuleService extends BlockTypeService<
...
+      });
+    }
+
+    return super.findMany(prismaArgs);
+  }
+
+  async create(
+    args: CreateModuleArgs,
+    user: User,
+    trackEvent = true
```
Comment (mulygottlieb) [inline packages/amplication-server/src/core/module/module.service.ts:94]: should it default to true or false?

File: packages/amplication-server/src/core/module/module.service.ts [hunk 2]
```diff
@@ -100,6 +145,20 @@ export class ModuleService extends BlockTypeService<
...
+
+    const subscription = await this.billingService.getSubscription(
+      user.workspace?.id
+    );
+
+    await this.analytics.trackWithContext({
+      properties: {
+        name: args.data.name,
+        planType: subscription.subscriptionPlan,
+        operation: "rename",
```
Comment (mulygottlieb) [inline packages/amplication-server/src/core/module/module.service.ts:157]: Is this the only possible update for Module (only renaming)?

File: packages/amplication-server/src/core/moduleAction/moduleAction.service.ts [hunk 3]
```diff
@@ -175,6 +222,19 @@ export class ModuleActionService extends BlockTypeService<
...
+      user.workspace?.id
+    );
+
+    await this.analytics.trackWithContext({
+      properties: {
+        name: moduleAction.name,
+        operation: "delete",
+        planType: subscription.subscriptionPlan,
+      },
+      event: EnumEventType.InteractUserAction,
```
Comment (mulygottlieb) [inline packages/amplication-server/src/core/moduleAction/moduleAction.service.ts:236]: Do you know this is never on an implication action here (e.g. use InteractAmplicationAction in some cases)?

File: packages/amplication-server/src/core/module/module.service.ts (hunk 1 above)
Comment (morhag90) [reply packages/amplication-server/src/core/module/module.service.ts:94]: true, sending false from the inner function

File: packages/amplication-server/src/core/module/module.service.ts (hunk 2 above)
Comment (morhag90) [reply packages/amplication-server/src/core/module/module.service.ts:157]: name and description @mulygottlieb

**Review by mulygottlieb — APPROVED** [review]

**Pull request merged**